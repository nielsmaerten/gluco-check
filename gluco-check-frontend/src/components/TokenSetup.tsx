import React from "react";
import { Box, makeStyles, Typography } from "@material-ui/core";
import { Trans, useTranslation } from "react-i18next";
import { Link } from "react-router-dom";

const useStyles = makeStyles((theme) => ({
  root: {
    maxWidth: "100%",
  },
  definitionList: {
    "& dd, & dt": {
      margin: "0px",
      padding: "0px",
    },
    "& dt": {
      fontWeight: "bold",
    },
    "& dd": {
      marginBottom: "1rem",
    },
    "& dd img": {
      maxWidth: "100%",
    },
  },
  button: {
    marginTop: theme.spacing(1),
    marginRight: theme.spacing(1),
  },
  actionsContainer: {
    marginBottom: theme.spacing(2),
  },
  resetContainer: {
    padding: theme.spacing(3),
  },
}));

const TOKEN_IMAGE_PATH = `${process.env.PUBLIC_URL}/images/token-instructions`;

export default function TokenSetup() {
  const { t } = useTranslation();
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <Typography>{t("tokenDialog.preamble")}</Typography>
      <dl className={classes.definitionList}>
        <Typography variant="h5" component="dt">
          {t("tokenDialog.step1.label")}
        </Typography>
        <dd>
          <Trans i18nKey="tokenDialog.step1.description">
            Make sure your site is on at least{" "}
            <a href={t("tokenDialog.links.nightscoutMinimumVersion")}>
              version 14 of Nightscout
            </a>
            . You can check the version here by clicking on the hamburger menu
            button and looking here...
          </Trans>
          <img
            src={`${TOKEN_IMAGE_PATH}/${t("tokenDialog.step1.image")}`}
            alt=""
          />
        </dd>

        <Typography variant="h5" component="dt">
          {t("tokenDialog.step2.label")}
        </Typography>
        <dd>
          <Trans i18nKey="tokenDialog.step2.description">
            If you need an upgrade, you can read more about how to do that{" "}
            <a href={t("tokenDialog.links.howToUpdateNightscout")}>here</a>. If
            you do not need an upgrade... let's goooooo!
          </Trans>
        </dd>

        <Typography variant="h5" component="dt">
          {t("tokenDialog.step3.label")}
        </Typography>
        <dd>
          <p>{t("tokenDialog.step3.description")}</p>
          <p>{t("tokenDialog.step4.description")}</p>
          <Box>
            <img
              src={`${TOKEN_IMAGE_PATH}/${t("tokenDialog.step4.image")}`}
              alt={t("tokenDialog.step4.image_alt")}
            />
          </Box>
          <p>{t("tokenDialog.step5.description")}</p>
          <img
            src={`${TOKEN_IMAGE_PATH}/${t("tokenDialog.step5.image")}`}
            alt={t("tokenDialog.step5.image_alt")}
          />
          <p>
            <Trans i18nKey="tokenDialog.step6.description">
              Copy the value in the "Access Token" column, paste it into the
              Nightscout Token field in{" "}
              <Link to={t("tokenDialog.links.glucoCheckSettings") as string}>
                Gluco-Check Settings
              </Link>
              , and click save.
            </Trans>
          </p>
          <img
            src={`${TOKEN_IMAGE_PATH}/${t("tokenDialog.step6.image")}`}
            alt={t("tokenDialog.step6.image_alt")}
          />
        </dd>
      </dl>
    </div>
  );
}
