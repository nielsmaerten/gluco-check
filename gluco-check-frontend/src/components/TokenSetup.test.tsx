import React from "react";
import { cleanup, render } from "@testing-library/react";
import { axe } from "jest-axe";
import TokenSetup from "./TokenSetup";

afterEach(() => {
  cleanup();
});

jest.mock("react-i18next", () => ({
  useTranslation: () => {
    return {
      t: jest.fn().mockImplementation((i) => {
        return i;
      }),
    };
  },
  Trans: function (props: any) {
    return <span>{props.i18nKey}</span>;
  },
}));

describe("TokenSetup component", () => {
  it("renders the component", () => {
    const { container } = render(<TokenSetup />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("has no axe violations", async () => {
    const { container } = render(<TokenSetup />);
    expect(await axe(container)).toHaveNoViolations();
  });
});
