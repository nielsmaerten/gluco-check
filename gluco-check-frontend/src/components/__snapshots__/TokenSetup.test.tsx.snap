// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TokenSetup component renders the component 1`] = `
<div
  class="makeStyles-root-1"
>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    tokenDialog.preamble
  </p>
  <dl
    class="makeStyles-definitionList-2"
  >
    <dt
      class="MuiTypography-root MuiTypography-h5"
    >
      tokenDialog.step1.label
    </dt>
    <dd>
      <span>
        tokenDialog.step1.description
      </span>
      <img
        alt=""
        src="/images/token-instructions/tokenDialog.step1.image"
      />
    </dd>
    <dt
      class="MuiTypography-root MuiTypography-h5"
    >
      tokenDialog.step2.label
    </dt>
    <dd>
      <span>
        tokenDialog.step2.description
      </span>
    </dd>
    <dt
      class="MuiTypography-root MuiTypography-h5"
    >
      tokenDialog.step3.label
    </dt>
    <dd>
      <p>
        tokenDialog.step3.description
      </p>
      <p>
        tokenDialog.step4.description
      </p>
      <div
        class="MuiBox-root MuiBox-root-6"
      >
        <img
          alt="tokenDialog.step4.image_alt"
          src="/images/token-instructions/tokenDialog.step4.image"
        />
      </div>
      <p>
        tokenDialog.step5.description
      </p>
      <img
        alt="tokenDialog.step5.image_alt"
        src="/images/token-instructions/tokenDialog.step5.image"
      />
      <p>
        <span>
          tokenDialog.step6.description
        </span>
      </p>
      <img
        alt="tokenDialog.step6.image_alt"
        src="/images/token-instructions/tokenDialog.step6.image"
      />
    </dd>
  </dl>
</div>
`;
